app/com/linkedin/drelephant/spark/heuristics/ConfigurationHeuristic.scala

```diff
@@ -128,18 +135,29 @@ object ConfigurationHeuristic {
     lazy val serializer: Option[String] = getProperty(SPARK_SERIALIZER_KEY)
 
     lazy val serializerSeverity: Severity = serializer match {
-      case None => Severity.NONE
+      case None => Severity.MODERATE
       case Some(`serializerIfNonNullRecommendation`) => Severity.NONE
```

Comment:
I know this is not part of your PR, but it looks suspicious nonetheless: IIUC the third branch is unreachable because the pattern ``Some(`foo`)`` is equavalent to `Some(_)` modulo variable name. 

```scala
scala> Some(42) match {
     | case Some(`foo`) => "first"
     | case Some(_) => "second"
     | }
res1: String = first
```